services/app-api/handlers/reports/write.ts

```diff
@@ -0,0 +1,70 @@
+import handler from "../handler-lib";
+import { getReport } from "./get";
+import dynamoDb from "../../utils/dynamo/dynamodb-lib";
+import { hasPermissions } from "../../utils/auth/authorization";
+import {
+  NO_KEY_ERROR_MESSAGE,
+  UNAUTHORIZED_MESSAGE,
+} from "../../utils/constants/constants";
+import { StatusCodes, UserRoles } from "../../utils/types/types";
+
+export const writeReport = handler(async (event, context) => {
+  if (!hasPermissions(event, [UserRoles.STATE_USER])) {
+    return {
+      status: StatusCodes.UNAUTHORIZED,
+      body: UNAUTHORIZED_MESSAGE,
+    };
+  } else if (!event?.pathParameters?.reportId!) {
+    throw new Error(NO_KEY_ERROR_MESSAGE);
+  } else {
```

Comment:
the above blocks return and throw. The functionality below doesn't need to be wrapped in an else, you can reduce some complexity here.